tests/src/core/testqgsexpression.cpp

```diff
@@ -2148,6 +2148,173 @@ class TestQgsExpression: public QObject
+      f.setFields( fields, true );
+      f.setAttribute( "col1", QString( "test value" ) );
+      QStringList array;
+      array << "one" << "two";
+      f.setAttribute( "strings", array );
+
+      QgsExpressionContext context = QgsExpressionContextUtils::createFeatureBasedContext( f, QgsFields() );
+
+      QVariantList builderExpected;
+      QCOMPARE( QgsExpression( "array()" ).evaluate( &context ), QVariant( ) );
+      builderExpected << "hello";
+      QCOMPARE( QgsExpression( "array('hello')" ).evaluate( &context ), QVariant( builderExpected ) );
+      builderExpected << "world";
+      QCOMPARE( QgsExpression( "array('hello', 'world')" ).evaluate( &context ), QVariant( builderExpected ) );
+
+      QCOMPARE( QgsExpression( "array_length(\"strings\")" ).evaluate( &context ), QVariant( 2 ) );
+
+      QCOMPARE( QgsExpression( "array_contains(\"strings\", 'two')" ).evaluate( &context ), QVariant( true ) );
+      QCOMPARE( QgsExpression( "array_contains(\"strings\", 'three')" ).evaluate( &context ), QVariant( false ) );
+
+      QCOMPARE( QgsExpression( "array_find(\"strings\", 'two')" ).evaluate( &context ), QVariant( 1 ) );
+      QCOMPARE( QgsExpression( "array_find(\"strings\", 'three')" ).evaluate( &context ), QVariant( -1 ) );
+
+      QCOMPARE( QgsExpression( "array_get(\"strings\", 1)" ).evaluate( &context ), QVariant( "two" ) );
+      QCOMPARE( QgsExpression( "array_get(\"strings\", 2)" ).evaluate( &context ), QVariant() );
+      QCOMPARE( QgsExpression( "array_get(\"strings\", -1)" ).evaluate( &context ), QVariant() );
+
+      QStringList appendExpected = array;
+      appendExpected << "three";
+      QCOMPARE( QgsExpression( "array_append(\"strings\", 'three')" ).evaluate( &context ), QVariant( appendExpected ) );
+
+      QStringList prependExpected = array;
+      prependExpected.prepend( "zero" );
+      QCOMPARE( QgsExpression( "array_prepend(\"strings\", 'zero')" ).evaluate( &context ), QVariant( prependExpected ) );
+
+      QStringList insertExpected = array;
+      insertExpected.insert( 1, "one and a half" );
+      QCOMPARE( QgsExpression( "array_insert(\"strings\", 1, 'one and a half')" ).evaluate( &context ), QVariant( insertExpected ) );
+
+      QStringList removeAtExpected = array;
+      removeAtExpected.removeAt( 0 );
+      QCOMPARE( QgsExpression( "array_remove_at(\"strings\", 0)" ).evaluate( &context ), QVariant( removeAtExpected ) );
+
+      QStringList removeAllExpected;
+      removeAllExpected << "a" << "b" << "d";
+      QCOMPARE( QgsExpression( "array_remove_all(array('a', 'b', 'c', 'd', 'c'), 'c')" ).evaluate( &context ), QVariant( removeAllExpected ) );
+
+      QStringList concatExpected = array;
+      concatExpected << "a" << "b" << "c";
+      QCOMPARE( QgsExpression( "array_concat(\"strings\", array('a', 'b'), array('c'))" ).evaluate( &context ), QVariant( concatExpected ) );
+    }
```

Comment:
I'd like to see some tests here where the array/map functions are called with non array/map values